packages/builder/src/pages/builder/auth/login.svelte

```diff
@@ -57,75 +64,96 @@
 </script>
 
 <svelte:window on:keydown={handleKeydown} />
-<div class="login">
-  <div class="main">
-    <Layout>
-      <Layout noPadding justifyItems="center">
-        <img alt="logo" src={$organisation.logoUrl || Logo} />
-        <Heading textAlign="center">Sign in to {company}</Heading>
-      </Layout>
+
+<TestimonialPage>
+  <Layout gap="S" noPadding>
+    <Layout justifyItems="center" noPadding>
       {#if loaded}
-        <GoogleButton />
-        <OIDCButton oidcIcon={$oidc.logo} oidcName={$oidc.name} />
+        <img alt="logo" src={$organisation.logoUrl || Logo} />
       {/if}
-      <Divider noGrid />
-      <Layout gap="XS" noPadding>
-        <Body size="S" textAlign="center">Sign in with email</Body>
-        <Input label="Email" bind:value={username} />
-        <Input
+      <Heading size="M">Log in to Budibase</Heading>
+    </Layout>
+    <Layout gap="S" noPadding>
+      {#if loaded && ($organisation.google || $organisation.oidc)}
+        <FancyForm>
+          <OIDCButton oidcIcon={$oidc.logo} oidcName={$oidc.name} />
+          <GoogleButton />
+        </FancyForm>
+        <Divider />
+      {/if}
+      <FancyForm bind:this={form}>
+        <FancyInput
+          label="Your work email"
+          value={formData.username}
+          on:change={e => {
+            formData = {
+              ...formData,
+              username: e.detail,
+            }
+          }}
+          validate={() => {
```

Comment:
Same as before, all you need to do is just return error strings, then invoke form.validate() and use the result to know wether the form is valid. You don't need to keep track of any error variables or pass any `error` props into the fields unless they're custom errors (such as returned from an API or something).

This is my bad - I musn't have explained that clearly enough.